identify/README.md

```diff
@@ -0,0 +1,51 @@
+# Identify v1.0.0
+
+The identify protocol is used to identify a peer and its capabilities.
+
+The protocol works by opening a stream, using `/ipfs/id/1.0.0` as the protocol string.
+The peer being identified responds by returning an `Identify` message and closes the
+stream.
+
+## The Identify Message
+
+```protobuf
+message Identify {
+  optional string protocolVersion = 5;
+  optional string agentVersion = 6;
+  optional bytes publicKey = 1;
+  repeated bytes listenAddrs = 2;
+  optional bytes observedAddr = 4;
+  repeated string protocols = 3;
+}
+```
+
+### protocolVersion
+
+The protocol version identifies the family of protocols used by the peer.
+The current protocol version is `ipfs/0.1.0`; if the protocol does not match
+the protocol used by the initiating peer, then the connection is considered
+unusable and the peer must close the connection.
+
+### agentVersion
+
+This is a free-form string, identifying the implementation of the peer.
+
+### publicKey
+
+This is the public key of the peer, marshalled in binary form as specicfied
+in [peer-ids](../peer-ids).
+
+
+### listenAddrs
+
+These are the addresses on which the peer is listening as multi-addresses.
+
+### observedAddr
```

Comment:
There are two possibilities here: this can be either the stack of protocols the listener used in order to reach back the dialer, or the stack of protocols we think the dialer used to reach us.
For the sake of forward compatibility, I think it'd be a nice idea to define this more precisely, and I'd be in favour of the former.

Reply:
Added a `source` disambiguator, per discussion in meatspace.

Reply:
Per discussion I'm not satisfied with the wording, because I'd like to disambiguate what is returned when `p2p-circuit` is used, but I can't find any appropriate wording.

Reply:
We can add an example for circuit addresses specifically, as they seem to be the contentious issue.

Reply:
Added some examples, hopefully it's clearer now.

Reply:
yeah, and it's not really any better than the current wording.